File: src/main/java/com/jss/osiris/modules/quotation/service/FormaliteStatusService.java
package com.jss.osiris.modules.quotation.service;

import java.util.List;

import com.jss.osiris.modules.quotation.model.FormaliteStatus;

public interface FormaliteStatusService {
    public List<FormaliteStatus> getFormaliteStatus();

    public FormaliteStatus getFormaliteStatus(Integer id);

    public FormaliteStatus addOrUpdateFormaliteStatus(FormaliteStatus formaliteStatus);

    public FormaliteStatus getFormaliteStatusByCode(String code);
}

File: client/src/app/modules/quotation/services/formalite.status.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/services/appRest.service';
import { FormaliteStatus } from '../../quotation/model/FormaliteStatus';

@Injectable({
  providedIn: 'root'
})
export class FormaliteStatusService extends AppRestService<FormaliteStatus> {

  constructor(http: HttpClient) {
    super(http, "quotation");
  }

  getFormaliteStatus() {
    return this.getList(new HttpParams(), "formalite-status");
  }

  addOrUpdateFormaliteStatus(formaliteStatus: FormaliteStatus) {
    return this.addOrUpdate(new HttpParams(), "formalite-status", formaliteStatus, "Enregistré", "Erreur lors de l'enregistrement");
  }

  getFormaliteStatusByCode(status: FormaliteStatus[], code: string) {
    if (status)
      for (let statu of status)
        if (statu.code == code)
          return statu;
    return null;
  }
}
